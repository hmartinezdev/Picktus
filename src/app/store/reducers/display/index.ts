import display from './display-reducer';
export { IDisplayState } from './display.type';

export default display;
