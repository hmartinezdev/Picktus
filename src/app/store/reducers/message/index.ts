import message from './message-reducer';
export { IMessageState } from './message.type';

export default message;
