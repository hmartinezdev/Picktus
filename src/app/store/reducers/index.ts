import { combineReducers } from 'redux';
import display, { IDisplayState } from './display';
import message, { IMessageState } from './message';
import user, { IUserState } from './user';

export interface IReduxState {
  user: IUserState;
  display: IDisplayState;
  message: IMessageState;
}

export default combineReducers({
  display,
  message,
  user,
});
