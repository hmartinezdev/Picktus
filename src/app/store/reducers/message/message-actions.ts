import { ThunkAction } from 'redux-thunk';
import { IMessageState, IPicktusError, IPicktusMessage } from './message.type';

export type ThunkResult<R> = ThunkAction<R, IMessageState, undefined, DismissNotification>;

export enum TypeKeys {
  MESSAGE_DISMISS_NOTIFICATION = 'MESSAGE_DISMISS_NOTIFICATION',
  MESSAGE_DISMISS_SNACKBAR = 'MESSAGE_DISMISS_SNACKBAR',
  MESSAGE_DISPLAY_SNACKBAR = 'MESSAGE_DISPLAY_SNACKBAR',
  MESSAGE_DISPLAY_NOTIFICATION = 'MESSAGE_DISPLAY_NOTIFICATION',
}

export interface DisplaySnackBar {
  type: TypeKeys.MESSAGE_DISPLAY_SNACKBAR;
  message: IPicktusError | IPicktusMessage;
}

export const displaySnackBar = (message: IPicktusError | IPicktusMessage): DisplaySnackBar => ({
  message,
  type: TypeKeys.MESSAGE_DISPLAY_SNACKBAR,
});

export interface DisplayNotification {
  type: TypeKeys.MESSAGE_DISPLAY_NOTIFICATION;
  message: IPicktusError | IPicktusMessage;
}

export const DisplayNotification = (message: IPicktusError | IPicktusMessage): DisplayNotification => ({
  message,
  type: TypeKeys.MESSAGE_DISPLAY_NOTIFICATION,
});

export interface DismissNotification {
  type: TypeKeys.MESSAGE_DISMISS_NOTIFICATION;
}

export const dismissNotification = (): DismissNotification => ({
  type: TypeKeys.MESSAGE_DISMISS_NOTIFICATION,
});

export interface DismissSnackBar {
  type: TypeKeys.MESSAGE_DISMISS_SNACKBAR;
}

export const dismissSnackBar = (): DismissSnackBar => ({
  type: TypeKeys.MESSAGE_DISMISS_SNACKBAR,
});

export type MessageActions = DismissNotification | DismissSnackBar | DisplayNotification | DisplaySnackBar;
