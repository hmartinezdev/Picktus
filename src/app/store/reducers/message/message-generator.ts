import uuid from 'uuid';
import { IPicktusError, IPicktusMessage, PicktusMessageDisplay, PicktusMessageLevel } from './message.type';

export default function(
  text: string,
  level: PicktusMessageLevel,
  display: PicktusMessageDisplay
): IPicktusError | IPicktusMessage {
  return {
    display,
    id: uuid.v4(),
    level,
    text,
  };
}
