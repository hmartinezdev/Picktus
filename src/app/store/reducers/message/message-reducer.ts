import { ActionTypes } from '../action.type';
import { IMessageState } from './message.type';
import { PicktusMessageDisplay, PicktusMessageLevel } from './message.type';

export default function user(
  state: IMessageState = {
    notifications: [
      {
        display: PicktusMessageDisplay.NOTIFICATION,
        id: 'ok',
        level: PicktusMessageLevel.SUCCESS,
        text: 'erreur de test',
      },
      {
        display: PicktusMessageDisplay.NOTIFICATION,
        id: 'ok',
        level: PicktusMessageLevel.ERROR,
        text: 'erreur de test',
      },
      {
        display: PicktusMessageDisplay.NOTIFICATION,
        id: 'ok',
        level: PicktusMessageLevel.WARNING,
        text: 'erreur de test',
      },
      {
        display: PicktusMessageDisplay.NOTIFICATION,
        id: 'ok',
        level: PicktusMessageLevel.INFO,
        text: 'erreur de test',
      },
    ],
    snackbars: [],
  },
  action: ActionTypes
): IMessageState {
  switch (action.type) {
    default:
      return state;
  }
}
