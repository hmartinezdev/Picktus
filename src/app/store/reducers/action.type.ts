import { DisplayActions } from './display/display-actions';
import { MessageActions } from './message/message-actions';
import { UserActions } from './user/user-actions';

enum TypeKeys {
  OTHER_ACTION = '__any_other_action_type__',
}

interface OtherAction {
  type: TypeKeys.OTHER_ACTION;
}

export type ActionTypes = UserActions | DisplayActions | MessageActions | OtherAction;
