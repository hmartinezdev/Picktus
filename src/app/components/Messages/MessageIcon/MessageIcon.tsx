import ErrorSvg from '@assets/svg/error.svg';
import InfoSvg from '@assets/svg/info.svg';
import SuccessSvg from '@assets/svg/sucess.svg';
import WarningSvg from '@assets/svg/warning.svg';
import colors from '@constants/colors';
import { PicktusMessageLevel } from '@store/reducers/message';
import React, { Component } from 'react';

export interface IMessageIconProps {
  level: PicktusMessageLevel;
  width: number;
  height: number;
}

class MessageIcon extends Component<IMessageIconProps> {
  public static defaultProps = {
    height: 32,
    width: 32,
  };

  public render(): React.ReactFragment {
    const { level, ...rest } = this.props;

    let icon = null;

    switch (level) {
      case PicktusMessageLevel.ERROR:
        icon = <ErrorSvg className="icon" key="icon" {...rest} />;
        break;
      case PicktusMessageLevel.WARNING:
        icon = <WarningSvg className="icon" key="icon" {...rest} />;
        break;
      case PicktusMessageLevel.SUCCESS:
        icon = <SuccessSvg className="icon" key="icon" {...rest} />;
        break;
      default:
        icon = <InfoSvg className="icon" key="icon" {...rest} />;
        break;
    }

    return [
      icon,
      <style key="style" jsx>{`
        .icon {
          fill: ${colors.secondary};
          margin-right: 0.3rem;
          min-height: ${rest.height};
          min-width: ${rest.width};
        }
      `}</style>,
    ];
  }
}

export default MessageIcon;
