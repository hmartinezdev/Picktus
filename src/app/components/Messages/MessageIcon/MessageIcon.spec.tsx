import { PicktusMessageLevel } from '@store/reducers/message';
import { shallow } from 'enzyme';
import React from 'react';
import MessageIcon from './MessageIcon';

let wrapper;

const setup = (props = {}) => shallow(<MessageIcon level={PicktusMessageLevel.ERROR} {...props} />);

describe('<MessageIcon />', () => {
  beforeEach(() => {
    wrapper = setup();
  });

  test('should render properly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  test('should render with different icon', () => {
    wrapper = setup({ level: PicktusMessageLevel.ERROR });
    expect(wrapper).toMatchSnapshot();
    wrapper = setup({ level: PicktusMessageLevel.SUCCESS });
    expect(wrapper).toMatchSnapshot();
    wrapper = setup({ level: PicktusMessageLevel.WARNING });
    expect(wrapper).toMatchSnapshot();
    wrapper = setup({ level: PicktusMessageLevel.INFO });
    expect(wrapper).toMatchSnapshot();
  });
});
