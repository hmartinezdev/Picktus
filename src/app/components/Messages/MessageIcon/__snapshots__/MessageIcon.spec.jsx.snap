// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageIcon /> should render properly 1`] = `
Array [
  <Component
    className="icon"
    height={32}
    key="icon"
    width={32}
  />,
  <style
    jsx={true}
    key="style"
  >
    
        .icon {
          fill: #FAC9B8;
          margin-right: 0.3rem;
          min-height: 32;
          min-width: 32;
        }
      
  </style>,
]
`;

exports[`<MessageIcon /> should render with different icon 1`] = `
Array [
  <Component
    className="icon"
    height={32}
    key="icon"
    width={32}
  />,
  <style
    jsx={true}
    key="style"
  >
    
        .icon {
          fill: #FAC9B8;
          margin-right: 0.3rem;
          min-height: 32;
          min-width: 32;
        }
      
  </style>,
]
`;

exports[`<MessageIcon /> should render with different icon 2`] = `
Array [
  <Component
    className="icon"
    height={32}
    key="icon"
    width={32}
  />,
  <style
    jsx={true}
    key="style"
  >
    
        .icon {
          fill: #FAC9B8;
          margin-right: 0.3rem;
          min-height: 32;
          min-width: 32;
        }
      
  </style>,
]
`;

exports[`<MessageIcon /> should render with different icon 3`] = `
Array [
  <Component
    className="icon"
    height={32}
    key="icon"
    width={32}
  />,
  <style
    jsx={true}
    key="style"
  >
    
        .icon {
          fill: #FAC9B8;
          margin-right: 0.3rem;
          min-height: 32;
          min-width: 32;
        }
      
  </style>,
]
`;

exports[`<MessageIcon /> should render with different icon 4`] = `
Array [
  <Component
    className="icon"
    height={32}
    key="icon"
    width={32}
  />,
  <style
    jsx={true}
    key="style"
  >
    
        .icon {
          fill: #FAC9B8;
          margin-right: 0.3rem;
          min-height: 32;
          min-width: 32;
        }
      
  </style>,
]
`;
